### PR TITLE
DynamicLoader: Resolve missing weak absolute symbols to null

### PR DESCRIPTION
**Pull request by IdanHo**
Weak symbols can be left unresolved, and code is supposed to check if they're null at runtime before using them. Handle them appropriately instead of refusing to load.
Together with #24072 we now succesfully boot to a graphical desktop on RISC-V :slightly_smiling_face:

### CODE REVIEW

**Review by DanShaders — APPROVED** [review]
lgtm

File: Userland/Libraries/LibELF/DynamicLoader.cpp [hunk 1]
```diff
@@ -569,21 +569,26 @@ DynamicLoader::RelocationResult DynamicLoader::do_direct_relocation(DynamicObjec
...
         auto symbol = relocation.symbol();
         auto res = lookup_symbol(symbol);
+        VirtualAddress symbol_address;
         if (!res.has_value()) {
-            if (symbol.bind() == STB_WEAK)
-                return RelocationResult::ResolveLater;
-            dbgln("ERROR: symbol not found: {}.", symbol.name());
-            return RelocationResult::Failed;
+            if (symbol.bind() == STB_WEAK) {
+                if (should_initialize_weak == ShouldInitializeWeak::No)
```
Comment (spholz) [inline Userland/Libraries/LibELF/DynamicLoader.cpp:575]: You should be able to remove the [[maybe_unused]] from should_initialize_weak now.

**Comment by warpdesign** [comment]
> Weak symbols can be left unresolved, and code is supposed to check if they're null at runtime before using them. Handle them appropriately instead of refusing to load. Together with #24072 we now succesfully boot to a graphical desktop on RISC-V 🙂

Video? :)

**Review by BertalanD — APPROVED** [review]
LGTM; but this part of DynamicLoader is in some serious need of refactoring.

**Comment by IdanHo** [comment]
> Video? :)

@warpdesign sure, it's quite slow since it's without KVM, but here you go:

https://github.com/SerenityOS/serenity/assets/16208640/dcfc9391-ca34-45b7-a4e0-6183ca7843f0

**Pull request merged**